ykrt/src/compile/j2/x64/x64hir_to_asm.rs

```diff
@@ -3052,8 +3055,8 @@ impl HirToAsmBackend for X64HirToAsm<'_> {
                 [
                     RegCnstr::InputOutput {
                         in_iidx: *lhs,
-                        in_fill: RegCnstrFill::Zeroed,
-                        out_fill: RegCnstrFill::Zeroed,
+                        in_fill: RegCnstrFill::Undefined,
```

Comment:
I read your other commit, but I'm still slightly confused why this works? Could you explain please?